src/createServer.js

```diff
@@ -1,3 +1,106 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+
+const http = require('http');
+
+function createServer() {
+  // Створення сервера
+  const server = http.createServer((req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+    const textToConvert = url.pathname.slice(1);
+    const toCase = url.searchParams.get('toCase');
+
+    const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    const errors = [];
+
+    if (!textToConvert) {
+      errors.push({
+        message:
+          'Text to convert is required. Correct request is: ' +
+          '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!toCase) {
+      errors.push({
+        message:
+          '"toCase" query param is required. Correct request is: ' +
+          '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    } else if (!supportedCases.includes(toCase)) {
+      errors.push({
+        message:
+          'This case is not supported. Available cases: ' +
+          'SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length > 0) {
+      res.writeHead(400, { 'Content-Type': 'application/json' });
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    let convertedText;
+    let originalCase;
+
+    if (/^[A-Z0-9_]+$/.test(textToConvert)) {
+      originalCase = 'UPPER';
+    } else if (/^[a-z0-9_]+$/.test(textToConvert)) {
+      originalCase = 'SNAKE';
+    } else if (/^[a-z0-9-]+$/.test(textToConvert)) {
+      originalCase = 'KEBAB';
+    } else if (/^[a-z][a-zA-Z0-9]+$/.test(textToConvert)) {
+      originalCase = 'CAMEL';
+    } else if (/^[A-Z][a-zA-Z0-9]+$/.test(textToConvert)) {
+      originalCase = 'PASCAL';
+    } else {
+      originalCase = 'UNKNOWN';
```

Comment:
The regular expressions for detecting the original case of the text might not cover all edge cases. For instance, the 'UPPER' case regex assumes that all characters are uppercase and separated by underscores, which might not always be true. Consider revising these expressions to handle more variations or document the expected input format.